src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java

```diff
@@ -43,21 +70,87 @@ private MetadataDataAccessUtils() {}
 
   /** Generate an {@link AccessInfoModel} from a Snapshot */
   public static AccessInfoModel accessInfoFromSnapshot(final Snapshot snapshot) {
-    return makAccessInfo(
+    return makeAccessInfoBigQuery(
         snapshot.getName(),
         snapshot.getProjectResource().getGoogleProjectId(),
         snapshot.getTables());
   }
 
   /** Generate an {@link AccessInfoModel} from a Dataset */
   public static AccessInfoModel accessInfoFromDataset(final Dataset dataset) {
-    return makAccessInfo(
-        BigQueryPdao.prefixName(dataset.getName()),
-        dataset.getProjectResource().getGoogleProjectId(),
-        dataset.getTables());
+    CloudPlatformWrapper cloudPlatformWrapper =
+        CloudPlatformWrapper.of(dataset.getDatasetSummary().getStorageCloudPlatform());
+
+    if (cloudPlatformWrapper.isGcp()) {
+      return makeAccessInfoBigQuery(
+          BigQueryPdao.prefixName(dataset.getName()),
+          dataset.getProjectResource().getGoogleProjectId(),
+          dataset.getTables());
+    } else if (cloudPlatformWrapper.isAzure()) {
+      BillingProfileModel profileModel = dataset.getDatasetSummary().getDefaultBillingProfile();
+      Optional<AzureStorageAccountResource> storageAccountResource =
+          resourceService.getStorageAccount(dataset, profileModel);
+      if (storageAccountResource.isPresent()) {
+        return makeAccessInfoAzure(
+            dataset.getName(), storageAccountResource.get(), dataset.getTables(), profileModel);
+      } else {
+        throw new AzureResourceNotFoundException("Storage account for dataset not found");
+      }
+    } else {
+      throw new IllegalArgumentException("Unrecognized cloud platform");
+    }
+  }
+
+  private static AccessInfoModel makeAccessInfoAzure(
+      final String datasetName,
+      final AzureStorageAccountResource storageAccountResource,
+      final List<? extends Table> tables,
+      final BillingProfileModel profileModel) {
+    AccessInfoModel accessInfoModel = new AccessInfoModel();
+
+    BlobContainerClientFactory targetDataClientFactory =
+        azureBlobStorePdao.getTargetDataClientFactory(
+            profileModel, storageAccountResource, ContainerType.METADATA, false);
+
+    // Given the sas token, rebuild a signed url
+    BlobSasTokenOptions options =
+        new BlobSasTokenOptions(
+            Duration.ofMinutes(15),
```

Comment:
Should this time be configurable from the environment, or are we ok with it being 15 minutes and not going to change it?

Reply:
Ultimately, we should have this be a dataset or snapshot or request option but we should have a followon story to make Sas token times congurable